docs/content/concepts/testing.mdx

```diff
@@ -193,13 +193,105 @@ def my_foo_resource(context):
 
 
 def test_op_resource_def():
-    context = build_op_context(
-        resources={"foo": my_foo_resource.configured({"my_str": "bar"})}
-    )
+    context = build_op_context(resources={"foo": my_foo_resource.configured({"my_str": "bar"})})
     assert op_requires_foo(context) == "found bar"
 ```
 
-### Testing job execution with configs
+Note that when directly invoking ops, io managers specified on inputs and outputs are not used.
+
+### Testing assets
+
+Similar to ops, <PyObject object="asset" decorator /> decorated functions can be directly invoked. Doing so will invoke the underlying op computation.
+
+A basic asset, with no dependencies:
+
+```python file=/concepts/ops_jobs_graphs/unit_tests.py startafter=start_test_basic_asset endbefore=end_test_basic_asset
+from dagster import asset
+
+
+@asset
+def basic_asset():
+    return 5
+
+
+# An example unit test for basic_asset.
+def test_basic_asset():
+    assert basic_asset() == 5
+```
+
+An asset with input dependencies:
+
+```python file=/concepts/ops_jobs_graphs/unit_tests.py startafter=start_test_input_asset endbefore=end_test_input_asset
+from dagster import asset
+
+
+@asset
+def asset_with_inputs(x, y):
+    return x + y
+
+
+# An example unit test for asset_with_inputs.
+def test_asset_with_inputs():
+    assert asset_with_inputs(5, 6) == 11
+```
+
+Any resources defined on your asset (either directly, or using <PyObject object="with_resources" />) will be picked up when invoking the asset. Simply provide a context using <PyObject object="build_op_context" />.
+
+```python file=/concepts/ops_jobs_graphs/unit_tests.py startafter=start_test_resource_asset endbefore=end_test_resource_asset
+from dagster import asset, resource, build_op_context, with_resources
+
+
+@asset(required_resource_keys={"service"})
+def asset_reqs_service(context):
+    service = context.resources.service
+    ...
+
+
+@resource
+def service():
+    ...
+
+
+# asset_with_service now has resource service specified.
+asset_with_service = with_resources([asset_reqs_service], {"service": service})[0]
+
+
+def test_asset_with_service():
+    # When invoking asset_with_service, service resource will
+    # automatically be used.
+    result = asset_with_service(build_op_context())
+    ...
+```
+
+Resources provided on <PyObject object="build_op_context" /> will override any resources provided directly on the asset. This means you can avoid using your prod resources when testing.
```

Comment:
```suggestion
Resources provided on <PyObject object="build_op_context" /> will override any resources provided directly on the asset. This means you can avoid using production resources when testing.
```

Other places throughout the docs use `production` - just trying to keep things consistent.